sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSessionReceiver.java

```diff
@@ -63,14 +63,23 @@
         this.receiveLink = receiveLink;
         this.lockContainer = new LockContainer<>(ServiceBusConstants.OPERATION_TIMEOUT);
 
-        receiveLink.setEmptyCreditListener(() -> 1);
+        receiveLink.setEmptyCreditListener(() -> 0);
 
         final Flux<ServiceBusMessageContext> receivedMessagesFlux = receiveLink
             .receive()
             .publishOn(scheduler)
             .doOnSubscribe(subscription -> {
                 logger.verbose("Adding prefetch to receive link.");
-                receiveLink.addCredits(prefetch);
+                if (prefetch > 0) {
+                    receiveLink.addCreditsInstantly(prefetch);
+                }
+            })
+            .doOnRequest(request -> {  // request is of type long.
+                if (prefetch == 0) {  //  add "request" number of credits
+                    receiveLink.addCreditsInstantly((int) request);
+                } else {  // keep total credits "prefetch" if prefetch is not 0.
+                    receiveLink.addCreditsInstantly(Math.max(0, prefetch - receiveLink.getCredits()));
```

Comment:
Just like we have `addCreditsInstantly` , adding locally , Does `getCredits` return from local? And if it is returning from remote, there could be short time when there is diff in local v/s remote .
Should there is equivalent `getCreditsInstantly` ?

Reply:
getCredits() should return from local. I don't see a log for traffic of getCredits.